spec/lib/appsignal/rack/event_handler_spec.rb

```diff
@@ -38,26 +39,32 @@ def on_error(error)
   end
 
   context "when used with ::Rack::Events" do
+    # Do not silence the deprecation warning about using it with the
+    # `Rack::Events` middleware.
     let(:event_handler_instance) { described_class.new }
 
     it "emits a warning about using it with Rack::Events" do
-      ::Rack::Events.new(nil, [event_handler_instance])
+      events = ::Rack::Events.new(rack_app, [event_handler_instance])
 
-      logs = capture_logs { on_start }
+      logs = capture_logs { events.call({}) }
       expect(logs).to contains_log(
         :warn,
         /Rack::Events is not compatible with streaming bodies./
       )
     end
   end
 
-  context "When used with ::Appsignal::Rack::Events" do
-    let(:event_handler_instance) { described_class.new }
+  context "When used via ::Appsignal::Rack::EventMiddleware" do
+    # Prevent `let(:event_handler_instance)` from initialising an
+    # additional `Appsignal::Rack::EventHandler` instance.
+    let(:event_handler_instance) { nil }
 
     it "does not emit a warning about using it with Rack::Events" do
-      Appsignal::Rack::Events.new(nil, [event_handler_instance])
+      expect(described_class).to receive(:new).and_call_original
+
+      event_middleware = Appsignal::Rack::EventMiddleware.new(rack_app)
 
-      logs = capture_logs { on_start }
+      logs = capture_logs { event_middleware.call({}) }
```

Comment:
Giving the variable a different name would also prevent a `Appsignal::Rack::EventHandler` from being initialized right?

`event_middleware` to `something_else`

Reply:
Yeah, something tripped me up here, and I thought this was fixing some behaviour I was seeing, but that's not true. The `let(...) { nil }` doesn't actually do anything, and the comment makes no sense.